### ISSUE TITLE
`.save()` & `delete()` feature for queried data

### ISSUE THREAD

**Issue by rigba** [prisma/prisma#15941]
## Problem

When using typeORM a feature I really liked was the ability to save and delete queried data.

## Suggested solution

In my mind the feature would look something like this

```js
const post = await prisma?. post.findFirst({ where: { id: 1} })

if (randomCondition) {
 post.message = "..."
} else {
 post.message = "...2"
}

post.save()

```

instead of something like

```js
const post = await prisma?. post.update({ where: { id: 1} , data: { message: randomCondition ? "..." : "...2"}}

```

the above example is quite simple, but it can get quite annoying if you're trying to make something like a product filter with lots of different conditions. I'm not sure if this is possible for you guys to make from an engineering point of view or that there's a simpler alternative that already exists that I'm not considering.

## Alternatives

null

## Additional context

From memory the way that TypeORM did this was with their [Base Entity](https://typeorm.delightful.studio/classes/_repository_baseentity_.baseentity.html#save-1) class.

**Comment by jamesie** [comment]
accidently posted this before i finished writing it give me a moment

**Comment by kevinvdburgt** [comment]
I don't think that is possible as the `post` is just a plain javascript object in your example.

You could try something like this:
```ts
const post = await prisma?. post.findFirst({ where: { id: 1} })

if (randomCondition) {
 post.message = "..."
} else {
 post.message = "...2"
}

await prisma?.post.update({
  where: { id: post.id },
  data: { message: post.message },
});
```

or take a look at [client-extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions) to create a helper function like `await prisma?.post.save(post);`
